src/vm-memory/src/bitmap.rs

```diff
@@ -78,6 +78,13 @@ impl Bitmap {
     pub fn is_empty(&self) -> bool {
         false
     }
+
+    /// Reset all bitmap bits to 0.
+    pub fn reset(&self) {
+        for n in 0..self.size {
+            self.map[n >> 6].fetch_and(0 << (n & 63), Ordering::SeqCst);
+        }
```

Comment:
Using `Release` ordering would change the semantics and feels like premature optimization. If weakening the memory consistency is necessary, it should be documented why it is safe.

Reply:
Since the operation also changes from `fetch_and` (read + update + write) to a simple `store` (write), is `SeqCst` still necessary?

The `Acquire` part is no longer necessary right? So that only leaves asking ourselves:
 - do we need all threads to see these operations sequentially consistent? Or is sequential ordering not relevant as long as update atomicity is upheld?

For the current Firecracker snapshot use-case, higher level program logic guarantees exclusive access to the emulation thread to this bitmap so I believe Ordering constraints are irrelevant.

For a possibly future live-snapshotting use-case where we'd keep vCPUs active, this becomes important, but still _strict sequential ordering_ doesn't look necessary, right?

For other use-cases I'm having a failure of imagination to come up with a scenario where sequential consistency is necessary when clearing this bitmap.

Disclaimer: I'm fine with keeping `Ordering::SeqCst` here since it's not in any hot-path whatsoever, but I wanted to take this opportunity to learn more about these ordering requirements and how to best use them.

Reply:
I think this is very relevant for usecases such as migrating a microvm. I would go for `store(0, Ordering::Release);`

Reply:
Done